xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -77,7 +77,10 @@ static boolean isResourceNameValid(String resourceName, String typeUrl) {
     return true;
   }
 
-  static String canonifyResourceName(String resourceName) {
+  /*
+   * Convert the XDSTP resource name to its canonical version.
+   */
+  public static String canonifyResourceName(String resourceName) {
```

Comment:
Can you merge from the master and mark this static method as experimental API. I am aware the class itself is marked as experimental too.
`@ExperimentalApi("https://github.com/grpc/grpc-java/issues/10862")`

Reply:
Done.